src/java.base/share/classes/java/util/Properties.java

```diff
@@ -1233,10 +1245,22 @@ public void list(PrintStream out) {
     }
 
     /**
-     * Prints this property list out to the specified output stream.
+     * Prints this property list out to the specified {@link PrintWriter}.
      * This method is useful for debugging.
      *
-     * @param   out   an output stream.
+     * @implNote The implementation of this method truncates the output for very long
+     * property values.
+     * <p>
+     * An alternative for listing the {@code Properties} to a {@code PrintWriter}
+     * without truncation is:
+     * {@snippet lang=java :
+     *      Properties p = ...
+     *      PrintWriter out = ...
+     *      // list the properties to PrintWriter
+     *      p.forEach((k, v) -> out.println(k + "=" + v));
```

Comment:
Should our example recommend using `toString` as a utility to obtain a debug string instead, as it performs no truncation?

Reply:
Using `toString()` will be harder to read; as proposed each property is on a separate line.